mathesar_ui/src/stores/databases.ts

```diff
@@ -19,19 +25,24 @@ export interface DatabaseStoreData {
   state: States;
   data: Database[];
   error?: string;
+  successfulConnections: SuccessfullyConnectedDatabase[];
 }
 
 export const databases = writable<DatabaseStoreData>({
   preload: true,
   state: States.Loading,
   data: commonData?.databases ?? [],
+  successfulConnections: (commonData?.databases ?? []).filter(
+    (database): database is SuccessfullyConnectedDatabase =>
+      !('error' in database),
+  ),
 });
 
 export const currentDatabase = derived(
   [currentDBName, databases],
   ([_currentDBName, databasesStore]) => {
     // eslint-disable-next-line @typescript-eslint/naming-convention
-    const _databases = databasesStore.data;
+    const _databases = databasesStore.successfulConnections;
```

Comment:
CRITICAL:

Please revert this change, and all places where we are only checking successfulConnections. This will break our application in many places if there's any unsuccessful connection. Always check all the databases, including the unsuccessful ones.  Make sure all components use `data` directly instead of `successfulConnections`.

An example:
* Have 2 DBs. Modify one DB, so that the connection is unsuccessful. The UI shows that correctly.
* Click on a different DB on the left pane, it shows it's page correctly.
* Click back on the unsuccessful DB. Notice a 404 page, instead of the error page for unsuccessful DB.
  * This happens because the DB route sets `currentDatabaseName` to the opened DB, which inturn modifies the `currentDatabase` which causes a range of ripple effects, because the usage of `successfulConnections` vs `data` is inconsistent.

I'm sure this also breaks in several other places causing race conditions, which wouldn't be immediately visible.